src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,1196 @@
+
+    // Verify the provided presetID is within spec limits
+    //
+    if ((preset > mMaxPresets) || (preset < 1))
+    {
+        ChipLogError(Zcl, "CameraAVSettingsUserLevelMgmt[ep=%d]: Preset provided is out of range. Preset: %d", mEndpointId, preset);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+        return;
+    }
+    // Do we have any presets?
+    //
+    if (mMptzPresetHelpers.empty())
+    {
+        ChipLogError(Zcl, "CameraAVSettingsUserLevelMgmt[ep=%d]: No stored presets, MoveToPreset not possible", mEndpointId);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return;
+    }
+
+    // We have presets, check that the received ID is a valid preset ID
+    //
+    auto it = std::find_if(mMptzPresetHelpers.begin(), mMptzPresetHelpers.end(),
+                           [preset](const MPTZPresetHelper & mptzph) { return mptzph.GetPresetID() == preset; });
+
+    if (it == mMptzPresetHelpers.end())
+    {
+        ChipLogError(Zcl,
+                     "CameraAVSettingsUserLevelMgmt[ep=%d]: No matching presets, MoveToPreset not possible for provided preset: %d",
+                     mEndpointId, preset);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return;
+    }
+
+    // This is effectively a manipulation of the current PTZ settings, ensure that the device is in a state wherein a PTZ change is
+    // possible
+    //
+    if (!mDelegate->CanChangeMPTZ())
+    {
+        ChipLogDetail(Zcl, "CameraAVSettingsUserLevelMgmt[ep=%d]: Device not able to process move to MPTZ preset", mEndpointId);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Busy);
+        return;
+    }
+
+    auto presetValues = it->GetMptzPosition();
+
+    // Inform the delegate that the device is requested to move to PTZ values given by the selected preset id
+    // Call the delegate to allow the device to handle the physical changes, on success set the MPTZ values based on the preset
+    status = mDelegate->MPTZMoveToPreset(preset, presetValues.pan, presetValues.tilt, presetValues.zoom);
+
+    if (status != Status::Success)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Set individual Pan, Tilt, and Zoom from the preset
+    SetPan(presetValues.pan);
+    SetTilt(presetValues.tilt);
+    SetZoom(presetValues.zoom);
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Success);
+}
+
+void CameraAvSettingsUserLevelMgmtServer::HandleMPTZSavePreset(HandlerContext & ctx,
+                                                               const Commands::MPTZSavePreset::DecodableType & commandData)
+{
+    Status status = Status::Success;
+
+    Optional<uint8_t> preset  = commandData.presetID;
+    chip::CharSpan presetName = commandData.name;
+    uint8_t presetToUse       = mCurrentPresetID;
+    bool newPresetValue       = true;
```

Comment:
Two things:

1.  Not sure why this is declared all the way up here, not where it used to be declared below where it's first computed for real.
2. The logic setting this value seems pretty complicated.  I wonder whether it would be better to reverse the sense of this boolean, call it `updatingExistingPreset` or something, and then `bool updatingExistingPreset = preset.HasValue() && it != mMptzPresetHelpers.end();`, which makes it pretty clear what's going on: a preset value was provided, and it exists.